plugins/inputs/windows_event_log/wineventlog/wineventlogrecord.go

```diff
@@ -78,3 +82,47 @@ func (record *windowsEventLogRecord) Value() (valueString string, err error) {
 func (record *windowsEventLogRecord) Timestamp() string {
 	return fmt.Sprint(record.System.TimeCreated.SystemTime.UnixNano())
 }
+
+type EventData struct {
+	Values []Data `xml:",any"`
+}
+
+type UserData struct {
+	Values []Data `xml:",any"`
+}
+
+// UserData has slightly different schema than EventData so that we need to overrid this
+// unmarshal function to get similar structure
+//
+// https://learn.microsoft.com/en-us/windows/win32/wes/eventschema-userdatatype-complextype
+// https://learn.microsoft.com/en-us/windows/win32/wes/eventschema-eventdatatype-complextype
+func (u *UserData) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	in := struct {
+		Values []Data `xml:",any"`
+	}{}
+
+	// Read tokens until we find the first StartElement then unmarshal it.
+	for {
+		t, err := d.Token()
+		if err != nil {
+			return err
+		}
+
+		if se, ok := t.(xml.StartElement); ok {
+			err = d.DecodeElement(&in, &se)
+			if err != nil {
+				return err
+			}
+
+			u.Values = in.Values
+			d.Skip()
+			break
+		}
+	}
+
+	return nil
+}
+
+type Data struct {
```

Comment:
super minor nitpick: Data is the plural for datum, so I think this would technically represent one `Datum`, not one `Data`.

Reply:
Thanks for calling this out. I've updated `Values` --> `Data` and `Data` --> `Datum`. This is making more sense.